lib/ui/painting.dart

```diff
@@ -1144,50 +1148,52 @@ enum Clip {
   ///
   /// See also:
   ///
-  ///  * [antiAlias], which is more reasonable when clipping is needed and the shape is not
+  ///  * [antiAlias], recommended when clipping is needed and the shape is not
   ///    an axis-aligned rectangle.
   hardEdge,
 
   /// Clip with anti-aliasing.
   ///
-  /// This mode has anti-aliased clipping edges to achieve a smoother look.
-  ///
-  /// It' s much faster than [antiAliasWithSaveLayer], but slower than [hardEdge].
+  /// This mode has anti-aliased clipping edges, which reduces jagged edges in
+  /// curves and rectangles that aren't axis-aligned.
```

Comment:
I'd recommend "reduces jagged edges when the clip shape itself has diagonal or curved edges or is transformed to not be axis-aligned" as "reduces ... in curves (plural?) and rectangles (plural?)" implies that it might magically improve the stuff rendered inside the clip - but really it is modifying only the edges of the clip shapes themselves.

Reply:
Ah yup I could see that being confusing, updated!